### PR TITLE
Add a simple cache of computed styles

### PR DESCRIPTION
**Pull request by kerrmarin**
Tests incoming

### CODE REVIEW

File: .github/workflows/unittestall.yml [hunk 1]
```diff
@@ -7,12 +7,13 @@ jobs:
...
+    - uses: maxim-lobanov/setup-xcode@v1.2.1
+      with:
+        xcode-version: 12.3
     - name: Test StylableSwiftUI
       shell: bash
       run: |
         cd Example
-        set -o pipefail && xcodebuild test -enableCodeCoverage YES -workspace StylableSwiftUI.xcworkspace -scheme StylableSwiftUI-Example ONLY_ACTIVE_ARCH=NO CODE_SIGN_IDENTITY="" CODE_SIGNING_REQUIRED=NO -destination 'platform=iOS Simulator,name=iPhone 11,OS=13.0' | xcpretty --color
+        set -o pipefail && xcodebuild test -enableCodeCoverage YES -workspace StylableSwiftUI.xcworkspace -scheme StylableSwiftUI-Example ONLY_ACTIVE_ARCH=NO CODE_SIGN_IDENTITY="" CODE_SIGNING_REQUIRED=NO -destination 'platform=iOS Simulator,name=iPhone 12,OS=14.3' | xcpretty --color
         cd ..
```
Comment (deanWombourne) [inline .github/workflows/unittestall.yml:18]: ^ This file can go directly into master if it works - it's nothing to do with caching ;)

Comment (deanWombourne) [reply .github/workflows/unittestall.yml:18]: Though, should we start . . . gulp . . . a matrix to test on various iOS versions?

**Review by deanWombourne — APPROVED** [review]
LGTM, pending tests (and performance tests maybe - though obviously a cache is going to be faster than calculating it each time :) Would be very interested to see how it actually affects a live app - should we add some debugging performance metrics to the library (opt-in obvs)?

The other approach to caching would be to cache the score caused by comparing identifiers - i.e. the cache key would be `(StylistIdentifier, StylistIdentifier)` - this would mean that we would still have to sort the array each time to get the best match, but we wouldn't have to clear it each time a style was added. In your app, do you ever add styles at 'runtime;' or do you set the stylist up at the start and then just use it?

**Comment by kerrmarin** [comment]
We set it up once at the start and that's it. In the future I'm not sure if we will do that exactly i.e. if we add dark mode support

**Comment by kerrmarin** [comment]
@deanWombourne we had some issues in the app where a list with several styled components was dropping frames while scrolling. Happy to add debugging metrics but not sure what exactly you want to measure?

**Comment by deanWombourne** [comment]
I was just thinking an average time to find a match, things like that. Just so you could easily see if/where slowdowns were occurring. Nor in this pr, but maybe add as an issue?

Anyway, if you've profiled it in your app and identified this is slow, and this fix fixes it then I'm happy 👍 

And if you don't dynamically add styles, let's not optimise for that. If this becomes a slowdown when you add dark mode then we can fix it then :)

**Comment by deanWombourne** [comment]
Oh, just thought, do we care how big this cache gets?

**Comment by kerrmarin** [comment]
@deanWombourne we're still dropping some frames, but profiler shows much less 👍 

This is a profiler screen from before this change while scrolling:

Screenshot 2020-12-21 at 16 38 15

And this is after (notice I scrolled twice as long this time):

Screenshot 2020-12-21 at 20 53 36

In the above, `count` is the number of times a body of a view was called, and the `Total Duration` is the number of seconds the app spent on those bodies. It's about 90% more efficient (if I've not made any mistakes measuring).

Next step is to somehow try to remove the double-executing in the body of `Stylable` if you have any ideas on how to tackle that :)

**Comment by kerrmarin** [comment]
re size of the cache - I haven't measured but I _don't think_ it'll be a problem. I'll take a look at measuring 100s and 1000s of styles and see what comes up in the profiler. Any ideas for a test to measure that?

File: .github/workflows/unittestall.yml (hunk 1 above)
Comment (kerrmarin) [reply .github/workflows/unittestall.yml:18]: I'll make a new PR with these changes

Comment (deanWombourne) [reply .github/workflows/unittestall.yml:18]: Honestly, for changes list this just make a branch and if the CI is happy just merge to master (assuming we haven't disabled that in github :)

**Pull request merged**